.github/workflows/check_pr.yml

```diff
@@ -0,0 +1,93 @@
+name: Check PR
+
+on: pull_request
+
+jobs:
+  build:
+    if: github.repository_owner == 'pnp'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+        node: [18]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          registry-url: "https://registry.npmjs.org"
+      - name: Cache node modules
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            **/node_modules
+          key: node_modules-${{ matrix.os }}-${{ matrix.node }}-${{ hashFiles('**/npm-shrinkwrap.json') }}
+      - name: Restore dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: npm ci
+      - name: Build
+        run: npm run build
+      - name: Compress output (non-Windows)
+        if: matrix.os != 'windows-latest'
+        run: tar -cvf build.tar --exclude node_modules ./
+      - name: Compress output (Windows)
+        if: matrix.os == 'windows-latest'
+        run: 7z a -ttar -xr!node_modules -r build.tar .
+      - name: Upload build artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-${{ matrix.os }}-${{ matrix.node }}
+          path: build.tar
```

Comment:
In `.github/workflows/test.yml` we have an empty line before a new task which looks quite neat. So a suggestion would be to apply it here as well 😄 

```suggestion
      - name: Use Node.js ${{ matrix.node }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node }}
          registry-url: "https://registry.npmjs.org"
          
      - name: Cache node modules
        id: cache
        uses: actions/cache@v3
        with:
          path: |
            **/node_modules
          key: node_modules-${{ matrix.os }}-${{ matrix.node }}-${{ hashFiles('**/npm-shrinkwrap.json') }}
          
      - name: Restore dependencies
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
        
      - name: Build
        run: npm run build
        
      - name: Compress output (non-Windows)
        if: matrix.os != 'windows-latest'
        run: tar -cvf build.tar --exclude node_modules ./
        
      - name: Compress output (Windows)
        if: matrix.os == 'windows-latest'
        run: 7z a -ttar -xr!node_modules -r build.tar .
        
      - name: Upload build artifact
        uses: actions/upload-artifact@v3
        with:
          name: build-${{ matrix.os }}-${{ matrix.node }}
          path: build.tar
```